<a href="https://colab.research.google.com/github/Kashika18/AI-Kickstart/blob/main/Insta_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [16]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Reading Data

In [17]:
train = pd.read_csv('train.csv')

In [18]:
insta = pd.DataFrame(train)

In [19]:
insta.head()

profile pic  nums/length username  ...  #follows  fake
0            1                  0.27  ...       955     0
1            1                  0.00  ...       533     0
2            1                  0.10  ...        98     0
3            1                  0.00  ...       651     0
4            1                  0.00  ...       126     0

[5 rows x 12 columns]

In [20]:
insta.info

<bound method DataFrame.info of      profile pic  nums/length username  ...  #follows  fake
0              1                  0.27  ...       955     0
1              1                  0.00  ...       533     0
2              1                  0.10  ...        98     0
3              1                  0.00  ...       651     0
4              1                  0.00  ...       126     0
..           ...                   ...  ...       ...   ...
571            1                  0.55  ...       596     1
572            1                  0.38  ...        75     1
573            1                  0.57  ...       339     1
574            1                  0.57  ...        73     1
575            1                  0.27  ...       487     1

[576 rows x 12 columns]>

# Analysing Data

In [21]:
insta['pp'] = insta['profile pic'][insta['profile pic']==0]

In [22]:
maximum_followers=insta.groupby(['pp'])['#followers'].max()

In [23]:
#answer 1
print(maximum_followers)

pp
0.0    3033
Name: #followers, dtype: int64



The maximum no. of followers possessed by a user belonging to the group of
users with no profile picture are **3033**

In [61]:
no_dp= insta.groupby(['profile pic']).get_group(0)

In [63]:
no_posts=no_dp.groupby(['#posts']).get_group(0)


In [64]:
no_bio=no_posts.groupby(['description length']).get_group(0)

In [68]:
#Total number of dead accounts
no_bio.shape[0]

123

Dead accounts are characterized by a lack of substantial activity on the platform,
primarily concerning the accounts with no profile picture, empty bio and 0 posts. 

**123 dead accounts exist.**

In [29]:
#finding average description length
avg=insta['description length'].mean()

In [88]:
print(avg)

22.62326388888889


In [30]:
insta['express']=['very expressive' if insta['description length'][i]>avg else 'less expressive' for i in insta.index]

An expressive person can is the user whose description_length exceeds
the average.

In [31]:

print(insta)

     profile pic  nums/length username  ...  posts          express
0              1                  0.27  ...    NaN  very expressive
1              1                  0.00  ...    NaN  very expressive
2              1                  0.10  ...    NaN  less expressive
3              1                  0.00  ...    NaN  very expressive
4              1                  0.00  ...    NaN  less expressive
..           ...                   ...  ...    ...              ...
571            1                  0.55  ...    NaN  less expressive
572            1                  0.38  ...    NaN  less expressive
573            1                  0.57  ...    NaN  less expressive
574            1                  0.57  ...    0.0  less expressive
575            1                  0.27  ...    NaN  less expressive

[576 rows x 16 columns]


In [60]:
#ans4
max_followers=insta.groupby(['posts'])['#followers'].max()

In [61]:
print(max_followers)

posts
0.0    3033
Name: #followers, dtype: int64


The maximum no. of followers possessed by a user among those who have 0
posts on their respective accounts are 3033

In [77]:
mx=insta['#follows'].max()

In [76]:
print(mx)

7500


The number of followers of the user with maximum number of following = 7500

In [103]:
data= pd.read_csv("train.csv", index_col="#follows")

In [105]:
first= data.loc[[7500]]

In [106]:
print(first)

          profile pic  nums/length username  ...  #followers  fake
#follows                                     ...                  
7500                1                   0.0  ...        1383     0
7500                1                   0.0  ...        1775     0

[2 rows x 11 columns]


The number of followers of the user with maximum number of following = 1775 and 1383

In [107]:
ratio= 1383/7500
ratio2=1775/7500

In [108]:
print(ratio)
ratio2

0.1844


0.23666666666666666

The ratio of number of followers (#followers) to number of following (#follows) for the user with the maximum number of following = 0.23666666666666666 and 0.1844